graylog2-server/src/main/java/org/graylog/plugins/views/search/engine/QueryBackend.java

```diff
@@ -149,4 +151,6 @@ default void validateQueryTimeRange(Query query, SearchConfig config) {
                     throw new IllegalTimeRangeException("Search out of allowed time range limit");
                 });
     }
+
+    SuggestionResponse suggest(SuggestionRequest req);
```

Comment:
It looks to me as if the feature and its implementations are distinct enough so that it does not need or benefit from coupling it to the `QueryBackend`. IMO it would benefit from being isolated in a separate class/interface.

Reply:
Good point, separated the query backend and suggestions